connect/transforms/src/main/java/org/apache/kafka/connect/transforms/field/SingleFieldPath.java

```diff
@@ -0,0 +1,585 @@
+            } else {
+                final SingleFieldPath fieldPath = new SingleFieldPath(field, version);
+                PATHS_CACHE.put(field, fieldPath);
+                return fieldPath;
+            }
+        }
+    }
+
+    SingleFieldPath(String pathText, FieldSyntaxVersion version) {
+        if (pathText == null || pathText.isEmpty()) { // empty path
+            this.path = new String[] {};
+        } else {
+            switch (version) {
+                case V1: // backward compatibility
+                    this.path = new String[] {pathText};
+                    break;
+                case V2:
+                    path = buildFieldPathV2(pathText);
+                    break;
+                default:
+                    throw new IllegalArgumentException("Unknown syntax version: " + version);
+            }
+        }
+    }
+
+    private String[] buildFieldPathV2(String pathText) {
+        // if no dots or wrapping backticks are used, then return path with single step
+        if (!pathText.contains(String.valueOf(DOT_CHAR))) {
+            return new String[] {pathText};
+        } else {
+            // prepare for tracking path steps
+            final List<String> steps = new ArrayList<>();
+            // avoid creating new string on changes
+            final StringBuilder s = new StringBuilder(pathText);
+
+            while (s.length() > 0) { // until path is traversed
+                // start processing backtick pair, if any
+                if (s.charAt(0) == BACKTICK_CHAR) {
+                    s.deleteCharAt(0);
+
+                    // find backtick closing pair
+                    int idx = 0;
+                    while (idx >= 0) {
+                        idx = s.indexOf(String.valueOf(BACKTICK_CHAR), idx);
+                        if (idx == -1) { // if not found, fail
+                            throw new IllegalArgumentException("Incomplete backtick pair at [...]`" + s);
+                        }
+                        // check that it is not escaped or wrapped in another backticks pair
+                        if (idx < s.length() - 1 // not wrapping the whole field path
+                                && (s.charAt(idx + 1) != DOT_CHAR // not wrapping
+                                || s.charAt(idx - 1) == BACKSLASH_CHAR)) { // ... or escaped
+                            idx++; // move index forward and keep searching
+                        } else { // it's the closing pair
+                            steps.add(escapeBackticks(s.substring(0, idx)));
+                            s.delete(0, idx + 2); // rm backtick and dot
+                            break;
+                        }
+                    }
+                } else { // process dots in path
+                    final int atDot = s.indexOf(String.valueOf(DOT_CHAR));
+                    if (atDot > 0) { // get path step and move forward
+                        steps.add(escapeBackticks(s.substring(0, atDot)));
```

Comment:
Aren't we guaranteed that the field name has no backticks here? Why the call to `escapeBackticks`?